hawkular-inventory-paths/src/main/java/org/hawkular/inventory/paths/Path.java

```diff
@@ -1208,17 +1212,17 @@ public Impl get() {
 
         @SuppressWarnings("unchecked")
         public This resource(String id) {
-            segments.add(new Segment(Resource.class, id));
+            segments.add(new Segment(SegmentType.r, id));
             return (This) this;
         }
 
-        public SDB data(Resources.DataRole role) {
-            segments.add(new Segment(DataEntity.class, role.name()));
+        public SDB data(String role) {
```

Comment:
K, I'm preparing a PR for this. Btw. there's a pending PR from me on your branch that makes this PR pass CI.